src/assets/js/access-groups.js

```diff
@@ -0,0 +1,326 @@
+import { Api, MembersApi } from './access-groups-api';
+
+const defaultGroupInvitationExpiration = 5;
+
+export default class AccessGroups {
+  constructor() {
+    this.api = new Api();
+    // TODO: Remove this once we can confirm the new members work
+    this.membersApi = new MembersApi();
+  }
+
+  /**
+   * Access Group Methods
+   */
+  async getGroup(groupName) {
+    try {
+      return await this.api.execute({
+        path: 'groups/get',
+        endpointArguments: [groupName],
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async updateGroupDetails(groupName, updateData) {
+    try {
+      return await this.api.execute({
+        path: 'groups/put',
+        endpointArguments: [groupName],
+        dataArguments: updateData,
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async leaveGroup(groupName) {
+    try {
+      return await this.api.execute({
+        path: 'self/delete',
+        endpointArguments: [groupName],
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async createGroup(form) {
+    try {
+      return await this.api.execute({
+        path: 'groups/post',
+        dataArguments: form,
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async closeGroup(groupName) {
+    try {
+      return await this.api.execute({
+        path: 'groups/delete',
+        endpointArguments: [groupName],
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  /**
+   * Members Methods
+   */
+
+  async getAllMembers(groupName) {
```

Comment:
This endpoint paginates and defaults to 20 members. The name might be misleading here. We should also add search and sorting options.